### PR TITLE
8832 use NR State - CONSUMED, EXPIRED

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue:* bcgov/entity#8832

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: client/src/components/application/Examine/ActionButtons.vue
Comment (severinbeauvais) [inline client/src/components/application/Examine/ActionButtons.vue:189]: You have removed the check for approved statuses. This function will now only return "true for approved" if there is an expiry date and the status is not Expired. Is this OK?

I'd like to see the comment on line 184 updated. Maybe use a whole-function comment explaining what conditions mean what. (So a future developer doesn't have to guess the business logic.) Thx.

File: client/src/components/application/Examine/ActionButtons.vue
Comment (cameron-freshworks) [inline client/src/components/application/Examine/ActionButtons.vue:187]: Clean!

Comment (cameron-freshworks) [reply client/src/components/application/Examine/ActionButtons.vue:187]: Is there a current state or nameRequest state enum?

Comment (cameron-freshworks) [reply client/src/components/application/Examine/ActionButtons.vue:187]: If not, it might be clean to implement one to minimize the strings in script. Your call tho!

File: client/src/components/application/Examine/RequestInfoHeader.vue
Comment (severinbeauvais) [inline client/src/components/application/Examine/RequestInfoHeader.vue:381]: I think you also neeed to allow for CONDITIONAL APPROVED here.

File: client/test/unit/specs/transactions.spec.js
Comment (cameron-freshworks) [inline client/test/unit/specs/transactions.spec.js:38]: Looks good!

**Review by cameron-freshworks — APPROVED** [review]
Looks good, just a minor question/comment/suggestion

File: client/src/components/application/Examine/RequestInfoHeader.vue
Comment (severinbeauvais) [inline client/src/components/application/Examine/RequestInfoHeader.vue:514]: Same comment as above: confirm conditions and update/add a comment please.

File: client/test/jest/ComponentSnapshotTests/application/examine/Recipe/conflicts/conflictInfoType/__snapshots__/NamesMatch.spec.js.snap
Comment (severinbeauvais) [inline client/test/jest/ComponentSnapshotTests/application/examine/Recipe/conflicts/conflictInfoType/__snapshots__/NamesMatch.spec.js.snap:143]: Please mention this change in the PR description.

File: client/test/unit/specs/transactions.spec.js
Comment (severinbeauvais) [inline client/test/unit/specs/transactions.spec.js:51]: Are there tests for EXPIRED and CONDITION? Do they still work OK? If not, please add some. Thx!

File: client/src/components/application/Examine/ActionButtons.vue
Comment (kialj876) [reply client/src/components/application/Examine/ActionButtons.vue:187]: that might lead into quite a lot of work for him 😛 , but I agree it would be nice

**Review by kialj876 — APPROVED** [review]
looks good to me. Nothing to add on top of what cam/sev already mentioned

File: client/src/components/application/Examine/ActionButtons.vue
Comment (cameron-freshworks) [reply client/src/components/application/Examine/ActionButtons.vue:187]: Absolutely, don't need to go too crazy. If there is some low hanging fruit giver, but no blocker from me :)

File: client/test/unit/specs/transactions.spec.js
Comment (vysakh-menon-aot) [reply client/test/unit/specs/transactions.spec.js:51]: This test is for consumed state. Test passes now with this change

File: client/src/components/application/Examine/RequestInfoHeader.vue
Comment (vysakh-menon-aot) [reply client/src/components/application/Examine/RequestInfoHeader.vue:521]: Comments added

File: client/src/components/application/Examine/ActionButtons.vue
Comment (vysakh-menon-aot) [reply client/src/components/application/Examine/ActionButtons.vue:193]: Comments added

File: client/src/components/application/Examine/ActionButtons.vue
Comment (vysakh-menon-aot) [reply client/src/components/application/Examine/ActionButtons.vue:191]: I thought of adding an enum for state but as kial mentioned here its quite a lot of work. So I just followed existing structure

File: client/test/unit/specs/transactions.spec.js
Comment (severinbeauvais) [reply client/test/unit/specs/transactions.spec.js:51]: Please add a test for Expired (if there isn't one already). Thanks.

Comment (vysakh-menon-aot) [reply client/test/unit/specs/transactions.spec.js:51]: Added

**Pull request merged**
